utils/wasm-gen/src/config/syscalls/injection.rs

```diff
@@ -0,0 +1,121 @@
+    Function(RangeInclusive<u32>),
+}
+
+/// Possible injection types for each sys-call.
+#[derive(Debug, Clone)]
+pub struct SysCallsInjectionTypes(HashMap<InvocableSysCall, SysCallInjectionType>);
+
+impl SysCallsInjectionTypes {
+    /// Instantiate a sys-calls map, where each gear sys-call is injected into wasm-module only once.
+    pub fn all_once() -> Self {
+        Self::new_with_injection_type(SysCallInjectionType::Function(1..=1))
+    }
+
+    /// Instantiate a sys-calls map, where no gear sys-call is ever injected into wasm-module.
+    pub fn all_never() -> Self {
+        Self::new_with_injection_type(SysCallInjectionType::None)
+    }
+
+    /// Instantiate a sys-calls map with given injection type.
+    fn new_with_injection_type(injection_type: SysCallInjectionType) -> Self {
+        let sys_calls = SysCallName::instrumentable();
+        Self(
+            sys_calls
+                .iter()
+                .cloned()
+                .map(|name| (InvocableSysCall::Loose(name), injection_type.clone()))
+                .chain(sys_calls.iter().cloned().filter_map(|name| {
+                    InvocableSysCall::has_precise_variant(name)
+                        .then_some((InvocableSysCall::Precise(name), injection_type.clone()))
+                }))
+                .collect(),
+        )
+    }
+
+    /// Gets injection type for given sys-call.
+    pub fn get(&self, name: InvocableSysCall) -> SysCallInjectionType {
+        self.0
+            .get(&name)
+            .cloned()
+            .expect("instantiated with all sys-calls set")
+    }
+
+    /// Sets possible amount range for the the sys-call.
+    pub fn set(&mut self, name: InvocableSysCall, min: u32, max: u32) {
+        self.0
+            .insert(name, SysCallInjectionType::Function(min..=max));
+
+        if let InvocableSysCall::Precise(sys_call) = name {
+            let Some(required_imports) = InvocableSysCall::required_imports_for_sys_call(sys_call) else {
+                return;
+            };
+
+            for &sys_call_import in required_imports {
+                self.enable_sys_call(InvocableSysCall::Loose(sys_call_import));
+            }
+        }
+    }
+
+    /// Enables given sys-call.
+    pub(crate) fn enable_sys_call(&mut self, name: InvocableSysCall) {
+        if let Some(injection_type @ SysCallInjectionType::None) = self.0.get_mut(&name) {
+            *injection_type = SysCallInjectionType::Import;
+        }
+    }
+
+    /// Same as [`SysCallsInjectionTypes::set`], but sets amount ranges for multiple sys-calls.
+    pub fn set_multiple(
+        &mut self,
+        sys_calls_freqs: impl Iterator<Item = (InvocableSysCall, RangeInclusive<u32>)>,
+    ) {
+        for (name, range) in sys_calls_freqs {
+            let (min, max) = range.into_inner();
+            self.set(name, min, max);
+        }
+    }
+}
+
+impl Default for SysCallsInjectionTypes {
+    fn default() -> Self {
+        Self::all_once()
+    }
+}
```

Comment:
It's very implicit, and as I see we don't use this `Default` impl anywhere

Reply:
This is not my code. It was previously in `SysCallsInjectionAmounts`